engine/sources/gitdns/source.go

```diff
@@ -206,26 +206,33 @@ func (gs *gitSourceHandler) shaToCacheKey(sha string) string {
 }
 
 type authSecret struct {
-	token bool
-	name  string
+	token      bool
+	name       string
+	credHelper bool
 }
 
 func (gs *gitSourceHandler) authSecretNames() (sec []authSecret, _ error) {
 	u, err := url.Parse(gs.src.Remote)
 	if err != nil {
 		return nil, err
 	}
+
+	isBitbucket := u.Host == "bitbucket.org"
```

Comment:
Do we need to worry about any other hosts, like we do with GitHub enterprise? 🤔

Reply:
my host name could be different for azure devops based on a combination of the protocol with http or ssh, as well as the way dev.azure.com and visualstudio.com are both valid addresses (due to legacy url).

Reply:
Mmh, I do agree that we could miss some edge cases, but here it was changes that were part of https://github.com/dagger/dagger/pull/8778. Basically, the core git implementation was not passing the PAT properly, as bitbucket has another way of passing the auth tokens.

We could have some edge cases on github enterprise and some azure, but I added some tests https://github.com/grouville/dagger/blob/a924e0e9b054ece63086badc9c84142cd0bb11d0/core/integration/git_test.go. I tested azure, and github enterprise has already being used by users. I do think that this should be ok ; if it breaks we should then extend I agree 👍